services/src/main/scala/cromwell/services/metadata/MetadataService.scala

```diff
@@ -82,12 +82,14 @@ object MetadataService {
     def size: Int = events.size
   }
 
-  val MaximumMetadataWriteAttempts = 10
-  final case class PutMetadataAction(events: Iterable[MetadataEvent], maxAttempts: Int = MaximumMetadataWriteAttempts) extends MetadataWriteAction
-  final case class PutMetadataActionAndRespond(events: Iterable[MetadataEvent], replyTo: ActorRef, maxAttempts: Int = MaximumMetadataWriteAttempts) extends MetadataWriteAction
+  val MaximumMetadataActionAttempts = 10
+  final case class PutMetadataAction(events: Iterable[MetadataEvent], maxAttempts: Int = MaximumMetadataActionAttempts) extends MetadataWriteAction
+  final case class PutMetadataActionAndRespond(events: Iterable[MetadataEvent], replyTo: ActorRef, maxAttempts: Int = MaximumMetadataActionAttempts) extends MetadataWriteAction
 
   final case object ListenToMetadataWriteActor extends MetadataServiceAction with ListenToMessage
 
+  final case class DeleteMetadataAction(workflowId: WorkflowId, replyTo: ActorRef, maxAttempts: Int = MaximumMetadataActionAttempts) extends MetadataServiceAction
```

Comment:
I'm not convinced deletion benefits from retries... the failure modes for `deleteNonLabelMetadataEntriesForWorkflow` don't seem to be the sorts of things that will resolve themselves with time

Reply:
E.g., a random network glitch on attempt to reach the DB? Or DB being down for some amount of time and then being brought up?

Reply:
we're already retrying puts...

Reply:
If we accept the premise that the DB connection can be flakey (I've never seen evidence of this, but could be true) then I was thinking we should push DB retries down into something like `cromwell.database.slick.SlickDatabase#runActionInternal`. It already has error handling & manages timeouts.

That said I agree `DeleteMetadataAction` matches an existing pattern